specs/Cryptography/Types.md

```diff
@@ -0,0 +1,72 @@
+
+
+# signature.rs
+
+PublicKey/SecretKey types are used to represent keypairs for Ed25519 or Secp256k1.
+
+```rust
+pub enum PublicKey {
+    ED25519(ED25519PublicKey),
+    SECP256K1(Secp256K1PublicKey),
+}
+```
+```rust
+pub enum SecretKey {
+    ED25519(ED25519SecretKey),
+    SECP256K1(secp256k1::key::SecretKey),
+}
+```
+```rust
+pub enum Signature {
+    ED25519(ed25519_dalek::Signature),
+    SECP256K1(Secp256K1Signature),
+}
+```
```

Comment:
Still too much Rust, and also we should not reference any third-party library API in our specifications.